### ISSUE TITLE
IAsyncDisposable support

### ISSUE THREAD

**Issue by jeremyVignelles** [DotNetAnalyzers/IDisposableAnalyzers#199]
I didn't find any issue for tracking this, so here it is:

IDisposableAnalyzers should support `IAsyncDisposable` the same way as `IDisposable`

My current case is this one:

```cs
private Timer _timer;

public async Task ResetTimerAsync()
{
   if(_timer != null)
   {
      await _timer.DisposeAsync();
      _timer = null; // Warns with IDISP003: Dispose previous before re-assigning.
   }
}
```

But there are a lot of other cases, basically every analyzer warning is impacted : https://github.com/DotNetAnalyzers/IDisposableAnalyzers/tree/master/documentation

- The analyzer should warn if a IAsyncDisposable is created  but not properly disposed.
- The analyzer should not warn if Dispose() or DisposeAsync() is called.
- If both IDisposable and IAsyncDisposable are implemented, warnings should be triggered only once and calling either Dispose or DisposeAsync should be considered OK

**Comment by jeremyVignelles** [comment]
I have several places in my code that lead me to this issue. I removed the `#pragma warning disable` and everything went fine. I'm quite confident that we can close it now.
